### PR TITLE
Parse version from Cargo.toml to tag docker image

### LINKED ISSUE #5: Push docker versions
- Read the docker version from Cargo.toml, and push _the version_ + _latest_

### PR DESCRIPTION
**Pull request by tjtelan**
- Read crate version number from the top of the Cargo.toml
- Add additional tag to build_docker target
- Add additional docker push to push_docker target
Resolves #5

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,11 @@
+CRATE_VERSION=$(shell head -4 Cargo.toml | grep ^version | awk '{print $$3}' | tr -d \")
```
Comment (omid) [inline Makefile]: You can use the following command:

`cargo pkgid | cut -d# -f2 | cut -d: -f2`

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+
 build_docker:
-	docker build --no-cache -t omidmr/gitversion docker
+	docker build --no-cache -t omidmr/gitversion -t omidmr/gitversion:$(CRATE_VERSION) docker
 
 push_docker:
-	docker push omidmr/gitversion
+	docker push omidmr/gitversion:latest
```
Comment (omid) [inline Makefile]: I think the _latest_ tag doesn't exist at this stage or I'm wrong?!

File: Makefile (hunk 1 above)
Comment (tjtelan) [reply Makefile]: Nice. I didn't know about `cargo pkgid`.

I don't know my way around the `cut` command all that well, but for me, only part of this command was needed.

```
$ cargo pkgid | cut -d# -f2
0.1.0
```

File: Makefile (hunk 2 above)
Comment (tjtelan) [reply Makefile]: The way I was intending to go about it, if you've already run `make build_docker` and it has run successfully, then you should always have 2 resulting tags. `latest` and the `cargo pkgid` versions are simultaneously tagged.

I ran the make rule to confirm.

```
$ make build_docker
docker build --no-cache -t omidmr/gitversion -t omidmr/gitversion:0.1.0 docker
Sending build context to Docker daemon  2.048kB
Step 1/6 : FROM rust:1 as builder
[...]
Successfully tagged omidmr/gitversion:latest
Successfully tagged omidmr/gitversion:0.1.0
```

However, in the event that you've run this by accident, you could potentially update `latest` unintentionally.

What I mean is reversing the order of the `docker push` could serve as a gate for not accidentally pushing an older cached image. If attempting to push the pkgid tagged image fails, then latest won't be modified in the docker registry.

Thoughts?

File: Makefile (hunk 1 above)
Comment (omid) [reply Makefile]: Yes, but the one I put is safer.

When the folder name and binary name are different, the pkgid will be like:
`file:///home/somebody/gitversion#binary_name:0.1.1`

To test it, just rename the folder name or the package name.

File: Makefile (hunk 2 above)
Comment (omid) [reply Makefile]: OK, yes, you are right and the way you have in your mind is fine :+1:

**Comment by omid** [comment]
Thanks a lot.

**Pull request merged**